packages/web-components/src/components/rux-tooltip/rux-tooltip.tsx

```diff
@@ -0,0 +1,300 @@
+import {
+    Component,
+    Host,
+    h,
+    Prop,
+    Element,
+    Event,
+    EventEmitter,
+    Watch,
+    Method,
+    State,
+} from '@stencil/core'
+import {
+    Placement,
+    computePosition,
+    offset,
+    flip,
+    autoUpdate,
+    autoPlacement,
+} from '@floating-ui/dom'
+import { hasSlot } from '../../utils/utils'
+
+/**
+ * @slot (default) - The trigger for the rux-tooltip
+ * @part container - The container of the rux-tooltip text
+ * @part trigger-container - the container of the tooltip trigger
+ */
+
+export declare type ExtendedPlacement = Placement | 'auto'
+
+@Component({
+    tag: 'rux-tooltip',
+    styleUrl: 'rux-tooltip.scss',
+    shadow: true,
+})
+export class RuxTooltip {
+    private trigger!: HTMLElement
+    private content!: HTMLElement
+    private _positionerCleanup: ReturnType<typeof autoUpdate> | undefined
+
+    @Element() el!: HTMLRuxTooltipElement
+
+    @State() currentSlotted: any
+    @State() hasTriggerSlot = false
+    @State() delegatedFocus = false //Keeps track of whether the element in trigger slot is focusable
+
+    /**
+     *  Enter a string to be used as the tooltip on this element
```

Comment:
NIT: change to something like 'the tooltip's content'. 'string' in the description is a little redundant because the args table will already tell you the type

Reply:
Done